### DISCUSSION TITLE
ControlNet Not Working

### DISCUSSION THREAD

**Discussion by Josroyal** [TheLastBen/fast-stable-diffusion#1617] [General]
I'd heard that there was this thing called "ControlNet", a tool to control into detail poses, depth, etc. I wanted to use it but it doesn't work on the colab. I've already tried everything known: Reinstalling the colab on my gdrive more than 6 times, installing controlnet manually, using the safetensors and ckpt models, played with all the possible parameters, followed multiple online tutorials step by step and nothing works. It seems like the model doesn't detec the image because when i use the brush it detects the outlines of the brush but not the image. (Sorry if my english is not quite good, I'm a spanish speaker). Here are some pictures:
![Screenshot 2023-02-21 144810](https://user-images.githubusercontent.com/64459744/220444846-aae9e5dc-7dd7-4829-aa66-a6991c9201a5.jpg)
![Screenshot 2023-02-21 144822](https://user-images.githubusercontent.com/64459744/220444848-d0141d25-8547-4015-b1f1-e0b413127e08.jpg)
![Screenshot 2023-02-21 144704](https://user-images.githubusercontent.com/64459744/220444850-6bf8c91d-72a9-454c-90cb-b4b0cddf30fa.jpg)
![Screenshot 2023-02-21 144729](https://user-images.githubusercontent.com/64459744/220444852-6b7a9a5b-7a01-4b7b-a09b-50789ecc95ff.jpg)

**Comment by TheLastBen** [comment]
you just need to learn how to use it, it works fine, start with scribble

**Reply by Topzie** [reply]
It doesn't. It should use the image you put in the controlnet.

Have you noticed the error message it throws?

File "/content/gdrive/MyDrive/sd/stable-diffusion-webui/extensions/sd-webui-controlnet/scripts/controlnet.py", line 555, in postprocess
is_img2img_batch_tab = is_img2img and img2img_tab_tracker.submit_img2img_tab == 'img2img_batch_tab'
NameError: name 'img2img_tab_tracker' is not defined

**Comment by Josroyal** [comment]
I don't think so dude, I've just discovered a huggin face space for using the controlnet model and it worked there. At first I thought the pics I was using were low quality but now Idk what's causing the problem on the colab. (I already tried to use all the models and all of them don't work)
![Screenshot 2023-02-22 004901](https://user-images.githubusercontent.com/64459744/220534629-691c8667-bdbc-4047-8ee5-5cfb5a88ddb4.jpg)
![Screenshot 2023-02-22 004838](https://user-images.githubusercontent.com/64459744/220534632-7b6b1d01-f501-4e41-b896-5d32c5fc76e7.jpg)

**Reply by Topzie** [reply]
You can use it in the img2img , but it only takes the img2img image as input. (which is wrong).

**Comment by meunumerotim333** [comment]
I use a lote the ControNET pose that respect the character face. Use on Txt2image. Working great

**Reply by Josroyal** [reply]
It doesn't work either 
![9936bdd4-d818-4132 gradio live_](https://user-images.githubusercontent.com/64459744/220730989-c99ab5f1-f055-4d79-a7d2-2297e7059116.png)

**Reply by Josroyal** [reply]
and here using txt2img doesnt work either
![9936bdd4-d818-4132 gradio live_ (1)](https://user-images.githubusercontent.com/64459744/220731749-93bb36d5-4b33-4e4e-a79a-2429666f4b8a.png)

**Reply by Topzie** [reply]
It only works when you put the image in the img2img field. So it works, but not as much as it should work.

**Reply by Josroyal** [reply]
> It only works when you put the image in the img2img field. So it works, but not as much as it should work.

It's true, it works with your method. Hope it can be fixed soon!
![b5056524-2229-4598 gradio live_](https://user-images.githubusercontent.com/64459744/220750636-13d5b937-1d74-4669-8fa1-9471f75fc578.png)

**Comment by MasterDenis** [comment]
I noticed the same exact issue with the others, so @TheLastBen if you can take a look it would be wonderful, thanks

**Reply by TheLastBen** [reply]
does scribble work ?

**Reply by cerarslan** [reply]
> does scribble work ?

Scribble works, sir. Pose doesn't.

**Reply by TheLastBen** [reply]
did you check with the extension repo ? maybe there are similar issues https://github.com/Mikubill/sd-webui-controlnet

**Reply by cerarslan** [reply]
> did you check with the extension repo ? maybe there are similar issues https://github.com/Mikubill/sd-webui-controlnet

Pose are working too, now but in the img2img section not CN. Do you manage to use multinet-model, sir?

**Comment by Topzie** [comment]
It's a browser issue. It works in Chrome.

https://github.com/Mikubill/sd-webui-controlnet/issues/302

**Reply by cerarslan** [reply]
> It's a browser issue. It works in Chrome.
> 
> [Mikubill/sd-webui-controlnet#302](https://github.com/Mikubill/sd-webui-controlnet/issues/302)
Thank you! I used brave but on chrome is working rn.

**Reply by Josroyal** [reply]
I was using Brave as well. Sorry @TheLastBen, everything is fine now.
